snapcraft/storeapi/__init__.py

```diff
@@ -437,6 +446,41 @@ def snap_release(self, snap_name, revision, channels):
 
         return response_json
 
+    def push_validation(self, snap_id, assertion):
+        data = {
+            'assertion': assertion.decode('utf-8'),
+        }
+        auth = _macaroon_auth(self.conf)
+        response = self.put(
+            'snaps/{}/validations'.format(snap_id), data=json.dumps(data),
+            headers={'Authorization': auth,
+                     'Content-Type': 'application/json',
+                     'Accept': 'application/json'})
+        if not response.ok:
+            raise errors.StoreValidationError(snap_id, response)
+        try:
+            response_json = response.json()
+        except (AttributeError, JSONDecodeError):
+            raise errors.StoreValidationError(snap_id, response)
```

Comment:
Yes please, the json decode error is just not friendly for a user.

What would help with debugging this later is to add a logger.debug line with `response` somewhere in the message. Like `logger.debug('Invalid response from the server when doing blah: {} {}'.format(response.code, response.text))`